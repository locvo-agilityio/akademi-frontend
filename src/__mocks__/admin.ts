export const MOCK_ADMIN = {
  id: 1,
  name: 'Nabila A',
  role: 'Admin',
  avatar: 'https://bit.ly/dan-abramov',
};
