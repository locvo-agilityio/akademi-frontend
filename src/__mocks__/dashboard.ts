import {
  CalendarIcon,
  FoodIcon,
  StudentIcon,
  TeacherIcon,
} from '@/components/icons';

export const MOCK_SUMMARY_DASHBOARD = [
  {
    title: 'Students',
    total: 932,
    bgIcon: 'primary',
    icon: StudentIcon,
  },
  {
    title: 'Teachers',
    total: 754,
    bgIcon: 'secondary',
    icon: TeacherIcon,
  },
  {
    title: 'Events',
    total: 40,
    bgIcon: 'yellow',
    icon: CalendarIcon,
  },
  {
    title: 'Foods',
    total: 32,
    bgIcon: 'darkBlue',
    icon: FoodIcon,
  },
];
