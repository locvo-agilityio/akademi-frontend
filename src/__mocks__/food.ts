export const MOCK_FOODS = [
  {
    name: 'Beef Steak with Fried Potato',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    image:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRCQPUcFz7CxVScB-zxL5pl2w6f4ntjw4mV-w&s',
  },
  {
    name: 'Beef Steak with Fried Potato New',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    image:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRCQPUcFz7CxVScB-zxL5pl2w6f4ntjw4mV-w&s',
  },
];
