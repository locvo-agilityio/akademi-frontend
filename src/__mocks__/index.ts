export * from './admin';
export * from './dashboard';
export * from './student';
export * from './pagination';
export * from './message';
export * from './food';
export * from './teacher';
export * from './schedule';
export * from './payment';
