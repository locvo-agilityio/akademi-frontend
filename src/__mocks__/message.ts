export const MOCK_MESSAGES = [
  {
    name: 'Samantha William',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Tony Soap',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Jordan Nico',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Nadila Adja',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
];
