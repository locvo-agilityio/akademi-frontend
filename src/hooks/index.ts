export * from './useStudent';
export * from './usePagination';
export * from './useTeacher';
export * from './useUpload';
