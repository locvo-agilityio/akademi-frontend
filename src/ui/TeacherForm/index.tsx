import { Button, Flex, HStack, Text, useToast, VStack } from '@chakra-ui/react';
import { Controller, useForm } from 'react-hook-form';
import { useActionState, useCallback, useEffect, useTransition } from 'react';
import { useQueryClient } from '@tanstack/react-query';
import { Link, useNavigate } from 'react-router-dom';

// Components
import {
  CustomToast,
  InputField,
  TextareaField,
  UploadImage,
} from '@/components';

// Constants
import {
  ADDITIONAL_TEACHER_INFO,
  BASIC_TEACHER_INFO,
  EDUCATION_TEACHER_INFO,
  ERROR_MESSAGES,
  PUBLIC_ROUTERS,
  SUCCESS_MESSAGES,
  teachersQueryKeys,
  TOAST_STATUS,
  VALIDATION_RULES,
} from '@/constants';

// Types
import { ITeacher } from '@/types';

// Actions
import { addTeacher, editTeacher } from '@/actions';

interface ITeacherFormProps {
  defaultValues?: ITeacher;
}

const TeacherForm = ({ defaultValues }: ITeacherFormProps) => {
  const toast = useToast();
  const queryClient = useQueryClient();
  const navigate = useNavigate();
  const [isPending, startTransition] = useTransition();
  const isEdit = !!defaultValues?.documentId;

  const actionForm = isEdit ? editTeacher : addTeacher;
  const [state, formAction] = useActionState(actionForm, undefined);

  const {
    firstName = '',
    lastName = '',
    phone = '',
    email = '',
    address = '',
    city = '',
    birthday = '',
    cityEducation = '',
    degree = '',
    description = '',
    endEducation = '',
    expertise = '',
    photo = '',
    startEducation = '',
    subject = '',
    university = '',
  } = defaultValues || {};

  const initialValues: ITeacher = {
    firstName: firstName,
    lastName: lastName,
    phone: phone,
    email: email,
    address: address,
    city: city,
    birthday: birthday,
    cityEducation: cityEducation,
    degree: degree,
    description: description,
    endEducation: endEducation,
    expertise: expertise,
    photo: photo,
    startEducation: startEducation,
    subject: subject,
    university: university,
  };

  const {
    control,
    setValue,
    getValues,
    formState: { isValid, isDirty },
  } = useForm<ITeacher>({
    mode: 'onBlur',
    reValidateMode: 'onChange',
    defaultValues: initialValues,
  });

  const handleFileChange = useCallback(
    (file: string) => {
      setValue('photo', file, { shouldDirty: true });
    },
    [setValue],
  );

  const actionSubmit = useCallback(
    async (formData: FormData) => {
      try {
        startTransition(() => {
          formData.append('photo', getValues('photo'));

          if (isEdit)
            formData.append('documentId', defaultValues?.documentId as string);

          formAction(formData);

          queryClient.invalidateQueries({
            queryKey: teachersQueryKeys.lists(),
          });

          toast({
            render: () => (
              <CustomToast
                status={TOAST_STATUS.SUCCESS}
                message={
                  isEdit
                    ? SUCCESS_MESSAGES.EDIT_TEACHER_SUCCESS
                    : SUCCESS_MESSAGES.ADD_TEACHER_SUCCESS
                }
              />
            ),
          });
        });
      } catch (err) {
        toast({
          render: () => (
            <CustomToast
              status={TOAST_STATUS.ERROR}
              message={
                isEdit
                  ? ERROR_MESSAGES.EDIT_TEACHER_FAILED
                  : ERROR_MESSAGES.ADD_TEACHER_FAILED
              }
            />
          ),
        });
      }
    },
    [defaultValues?.documentId, getValues, isEdit, queryClient, toast],
  );

  useEffect(() => {
    if (state) {
      queryClient.setQueryData(
        teachersQueryKeys.detail(state.data.documentId),
        state,
      );

      navigate(PUBLIC_ROUTERS.TEACHERS);
    }
  }, [state, queryClient, navigate]);

  return (
    <VStack as="form" alignItems="flex-end" w="full" action={actionSubmit}>
      <VStack mt={5} w="full" bgColor="white" borderRadius="2xl">
        <Text
          as="h3"
          w="full"
          px={10}
          h="60px"
          borderTopRadius="2xl"
          bgColor="primary"
          color="white"
          size="lg"
          fontWeight="bold"
          display="flex"
          alignItems="center"
        >
          Personal Details
        </Text>

        <HStack w="full" alignItems="flex-start" p={8} gap={10}>
          <Flex w="full" gap={7} direction="column" justifyContent="flex-start">
            {BASIC_TEACHER_INFO.map(
              ({
                name,
                label,
                placeholder,
                type,
                rules,
                ariaLabel,
                min,
                max,
              }) => (
                <Controller
                  key={name}
                  control={control}
                  rules={rules}
                  name={name}
                  render={({ field, fieldState: { error } }) =>
                    name === 'address' || name === 'description' ? (
                      <TextareaField
                        h={175}
                        label={label}
                        placeholder={placeholder}
                        {...field}
                        isError={!!error}
                        errorMessages={error?.message}
                        aria-label={ariaLabel}
                      />
                    ) : (
                      <InputField
                        label={label}
                        placeholder={placeholder}
                        type={type}
                        min={min}
                        max={max}
                        {...field}
                        isError={!!error}
                        errorMessages={error?.message}
                        aria-label={ariaLabel}
                      />
                    )
                  }
                />
              ),
            )}
          </Flex>

          <Flex w="full" gap={7} direction="column" justifyContent="flex-start">
            {ADDITIONAL_TEACHER_INFO.map(
              ({ name, label, placeholder, type, rules, ariaLabel }) => (
                <Controller
                  key={name}
                  control={control}
                  rules={rules}
                  name={name}
                  render={({ field, fieldState: { error } }) => (
                    <InputField
                      label={label}
                      placeholder={placeholder}
                      type={type}
                      {...field}
                      isError={!!error}
                      errorMessages={error?.message}
                      aria-label={ariaLabel}
                    />
                  )}
                />
              ),
            )}

            <UploadImage imageUrl={photo} onFileChange={handleFileChange} />

            <Controller
              key="city"
              control={control}
              rules={VALIDATION_RULES.CITY}
              name="city"
              render={({ field, fieldState: { error } }) => (
                <InputField
                  label="Place of Birth *"
                  placeholder="Please enter your city"
                  {...field}
                  isError={!!error}
                  errorMessages={error?.message}
                  aria-label="city"
                />
              )}
            />

            <Controller
              key="expertise"
              control={control}
              rules={VALIDATION_RULES.CITY}
              name="expertise"
              render={({ field, fieldState: { error } }) => (
                <InputField
                  label="Expertise *"
                  placeholder="Please enter your expertise"
                  {...field}
                  isError={!!error}
                  errorMessages={error?.message}
                  aria-label="expertise"
                />
              )}
            />
          </Flex>
        </HStack>
      </VStack>

      <VStack mt={5} w="full" bgColor="white" borderRadius="2xl">
        <Text
          as="h3"
          w="full"
          px={10}
          h="60px"
          borderTopRadius="2xl"
          bgColor="primary"
          color="white"
          size="lg"
          fontWeight="bold"
          display="flex"
          alignItems="center"
        >
          Education
        </Text>

        <HStack w="full" alignItems="flex-start" p={8} gap={10}>
          <Flex w="full" gap={7} direction="column" justifyContent="flex-start">
            <Controller
              key="university"
              control={control}
              rules={VALIDATION_RULES.UNIVERSITY}
              name="university"
              render={({ field, fieldState: { error } }) => (
                <InputField
                  label="University *"
                  placeholder="Please enter your university"
                  {...field}
                  isError={!!error}
                  errorMessages={error?.message}
                  aria-label="university"
                />
              )}
            />

            <Flex
              direction={{ base: 'column', xl: 'row' }}
              gap={6}
              alignItems="flex-end"
            >
              <Controller
                key="startEducation"
                control={control}
                rules={VALIDATION_RULES.DATE}
                name="startEducation"
                render={({ field, fieldState: { error } }) => (
                  <InputField
                    label="Start & End Date *"
                    type="date"
                    placeholder="Please enter your start education"
                    {...field}
                    isError={!!error}
                    errorMessages={error?.message}
                    aria-label="birthday"
                  />
                )}
              />

              <Controller
                key="city"
                control={control}
                rules={VALIDATION_RULES.DATE}
                name="endEducation"
                render={({ field, fieldState: { error } }) => (
                  <InputField
                    title="End Date"
                    type="date"
                    placeholder="Please enter your end education"
                    {...field}
                    isError={!!error}
                    errorMessages={error?.message}
                    aria-label="city"
                  />
                )}
              />
            </Flex>
          </Flex>

          <Flex w="full" gap={7} direction="column" justifyContent="flex-start">
            {EDUCATION_TEACHER_INFO.map(
              ({ name, label, placeholder, type, rules, ariaLabel }) => (
                <Controller
                  key={name}
                  control={control}
                  rules={rules}
                  name={name}
                  render={({ field, fieldState: { error } }) => (
                    <InputField
                      label={label}
                      placeholder={placeholder}
                      type={type}
                      {...field}
                      isError={!!error}
                      errorMessages={error?.message}
                      aria-label={ariaLabel}
                    />
                  )}
                />
              ),
            )}
          </Flex>
        </HStack>
      </VStack>

      <HStack mt={4} gap={3}>
        <Button
          as={Link}
          to={PUBLIC_ROUTERS.TEACHERS}
          w="full"
          p={7}
          variant="secondary"
          borderRadius="full"
        >
          Cancel
        </Button>

        <Button
          w="full"
          p={7}
          variant="primary"
          borderRadius="full"
          type="submit"
          isLoading={isPending}
          isDisabled={isPending || !isValid || !isDirty}
        >
          Submit
        </Button>
      </HStack>
    </VStack>
  );
};

export default TeacherForm;
