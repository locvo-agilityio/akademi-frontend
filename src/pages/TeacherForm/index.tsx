import { useParams } from 'react-router-dom';

// Components
import { Fallback } from '@/components';

// Hooks
import { useGetTeacher } from '@/hooks';

import { TeacherForm as TeacherFormUI } from '@/ui';

const TeacherForm = () => {
  const { id } = useParams();
  const { teacher, isTeacherLoading } = useGetTeacher(id ?? '');

  return isTeacherLoading && id ? (
    <Fallback />
  ) : (
    <TeacherFormUI defaultValues={teacher?.data} />
  );
};

export default TeacherForm;
