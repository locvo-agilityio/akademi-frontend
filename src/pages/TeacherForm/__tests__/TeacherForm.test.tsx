import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { BrowserRouter } from 'react-router-dom';

// Components
import TeacherForm from '..';

// Mocks
import { MOCK_TEACHERS } from '@/__mocks__';

// Hooks
import { useGetTeacher, useUploadImage } from '@/hooks';

const client = new QueryClient();

const mockHandleUploadImage = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(() => ({})),
  useNavigate: jest.fn(() => ({})),
}));

jest.mock('@/hooks', () => ({
  useUploadImage: jest.fn(),
  useGetTeacher: jest.fn(() => ({
    teacher: MOCK_TEACHERS[0],
  })),
}));

describe('TeacherForm', () => {
  beforeEach(() => {
    (useUploadImage as jest.Mock).mockReturnValue({
      handleUploadImage: mockHandleUploadImage,
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render successfully', () => {
    const { container } = render(
      <BrowserRouter>
        <QueryClientProvider client={client}>
          <TeacherForm />
        </QueryClientProvider>
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render fallbackUI when loading', () => {
    (useGetTeacher as jest.Mock).mockReturnValue({
      teacher: null,
      isTeacherLoading: true,
    });

    const { container } = render(
      <BrowserRouter>
        <QueryClientProvider client={client}>
          <TeacherForm />
        </QueryClientProvider>
      </BrowserRouter>,
    );

    expect(container).toBeInTheDocument();
  });
});
