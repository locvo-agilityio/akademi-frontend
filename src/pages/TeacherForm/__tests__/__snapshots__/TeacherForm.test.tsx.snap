// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherForm should render successfully 1`] = `
<div>
  <form
    action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')"
    class="chakra-stack css-1d0q5nm"
  >
    <div
      class="chakra-stack css-9m3hkh"
    >
      <h3
        class="chakra-text css-ux3uq4"
      >
        Personal Details
      </h3>
      <div
        class="chakra-stack css-es5q0n"
      >
        <div
          class="css-tfo7n4"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:r0:"
              id="field-:r0:-label"
            >
              First Name *
            </label>
            <input
              aria-label="firstName"
              class="chakra-input css-1lx593h"
              id="field-:r0:"
              name="firstName"
              placeholder="Please enter your first name"
              type="text"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:r1:"
              id="field-:r1:-label"
            >
              Email *
            </label>
            <input
              aria-label="email"
              class="chakra-input css-1lx593h"
              id="field-:r1:"
              name="email"
              placeholder="Please enter your email"
              type="email"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:r2:"
              id="field-:r2:-label"
            >
              Address
               *
            </label>
            <textarea
              aria-label="address"
              class="chakra-textarea css-wugvdb"
              id="field-:r2:"
              name="address"
              placeholder="Please enter your address"
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:r3:"
              id="field-:r3:-label"
            >
              Date of Birth *
            </label>
            <input
              aria-label="birthday"
              class="chakra-input css-1lx593h"
              id="field-:r3:"
              max="2000-12-31"
              min="1970-12-31"
              name="birthday"
              placeholder="Please enter your birthday"
              type="date"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:r4:"
              id="field-:r4:-label"
            >
              Subject
            </label>
            <input
              aria-label="subject"
              class="chakra-input css-1lx593h"
              id="field-:r4:"
              name="subject"
              placeholder="Please enter your subject"
              type="text"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:r5:"
              id="field-:r5:-label"
            >
              Description
               *
            </label>
            <textarea
              aria-label="description"
              class="chakra-textarea css-wugvdb"
              id="field-:r5:"
              name="description"
              placeholder="Please enter your description"
            />
          </div>
        </div>
        <div
          class="css-tfo7n4"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:r6:"
              id="field-:r6:-label"
            >
              Last Name *
            </label>
            <input
              aria-label="lastName"
              class="chakra-input css-1lx593h"
              id="field-:r6:"
              name="lastName"
              placeholder="Please enter your last name"
              type="text"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:r7:"
              id="field-:r7:-label"
            >
              Phone *
            </label>
            <input
              aria-label="phone"
              class="chakra-input css-1lx593h"
              id="field-:r7:"
              name="phone"
              placeholder="(xxx) xxx-xxxx"
              type="text"
              value=""
            />
          </div>
          <div
            class="chakra-stack css-enj3l8"
          >
            <label
              class="chakra-form__label css-1hm3o85"
            >
              Photo *
            </label>
            <div
              class="css-9t9a43"
            >
              <p
                class="chakra-text css-144ed6u"
              >
                Drag and drop or click here to select file
              </p>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <input
                  accept="image/*"
                  class="chakra-input css-13rbvqk"
                  hidden=""
                  id="field-:r8:"
                  title="Upload Image"
                  type="file"
                />
              </div>
            </div>
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:r9:"
              id="field-:r9:-label"
            >
              Place of Birth *
            </label>
            <input
              aria-label="city"
              class="chakra-input css-1lx593h"
              id="field-:r9:"
              name="city"
              placeholder="Please enter your city"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:ra:"
              id="field-:ra:-label"
            >
              Expertise *
            </label>
            <input
              aria-label="expertise"
              class="chakra-input css-1lx593h"
              id="field-:ra:"
              name="expertise"
              placeholder="Please enter your expertise"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-stack css-9m3hkh"
    >
      <h3
        class="chakra-text css-ux3uq4"
      >
        Education
      </h3>
      <div
        class="chakra-stack css-es5q0n"
      >
        <div
          class="css-tfo7n4"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:rb:"
              id="field-:rb:-label"
            >
              University *
            </label>
            <input
              aria-label="university"
              class="chakra-input css-1lx593h"
              id="field-:rb:"
              name="university"
              placeholder="Please enter your university"
              value=""
            />
          </div>
          <div
            class="css-vypz0o"
          >
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-1hm3o85"
                for="field-:rc:"
                id="field-:rc:-label"
              >
                Start & End Date *
              </label>
              <input
                aria-label="birthday"
                class="chakra-input css-1lx593h"
                id="field-:rc:"
                name="startEducation"
                placeholder="Please enter your start education"
                type="date"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <input
                aria-label="city"
                class="chakra-input css-1lx593h"
                id="field-:rd:"
                name="endEducation"
                placeholder="Please enter your end education"
                title="End Date"
                type="date"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="css-tfo7n4"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:re:"
              id="field-:re:-label"
            >
              Degree *
            </label>
            <input
              aria-label="degree"
              class="chakra-input css-1lx593h"
              id="field-:re:"
              name="degree"
              placeholder="Please enter your degree"
              type="text"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-1hm3o85"
              for="field-:rf:"
              id="field-:rf:-label"
            >
              City *
            </label>
            <input
              aria-label="cityEducation"
              class="chakra-input css-1lx593h"
              id="field-:rf:"
              name="cityEducation"
              placeholder="Please enter your city"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-stack css-lu0tkq"
    >
      <a
        class="chakra-button css-193qlt7"
        href="/teachers"
      >
        Cancel
      </a>
      <button
        class="chakra-button css-193qlt7"
        disabled=""
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
